src/helm/benchmark/presentation/run_display.py

```diff
@@ -0,0 +1,267 @@
+
+    train_trial_index: int
+    """Which replication"""
+
+    predicted_text: str
+    """Prediction text"""
+
+    truncated_predicted_text: Optional[str]
+    """The truncated prediction text, if truncation is required by the Adapter method."""
+
+    mapped_output: Optional[str]
+    """The mapped output, if an output mapping exists and the prediction can be mapped"""
+
+    reference_index: Optional[int]
+    """Which reference of the instance we're evaluating (if any)"""
+
+    stats: Dict[str, float]
+    """Statistics computed from the predicted output"""
+
+
+@dataclass(frozen=True)
+class DisplayRequest:
+    """
+    Captures a unit of evaluation for displaying in the web frontend.
+    """
+
+    # (instance_id, perturbation, train_trial_index) is a unique key for this prediction.
+    instance_id: str
+    """ID of the Instance"""
+
+    perturbation: Optional[PerturbationDescription]
+    """Description of the Perturbation that was applied"""
+
+    train_trial_index: int
+    """Which replication"""
+
+    request: Request
+    """The actual Request to display in the web frontend.
+
+    There can be multiple requests per trial. The displayed request should be the
+    most relevant request e.g. the request for the chosen cohice for multiple choice questions."""
+
+
+def _read_scenario_state(run_path: str) -> ScenarioState:
+    scenario_state_path: str = os.path.join(run_path, "scenario_state.json")
+    if not os.path.exists(scenario_state_path):
+        raise ValueError(f"Could not load ScenarioState from {scenario_state_path}")
+    with open(scenario_state_path) as f:
+        raw_scenario_state = json.load(f)
+        return dacite.from_dict(ScenarioState, raw_scenario_state, config=_DACITE_CONFIG)
+
+
+def _read_per_instance_stats(run_path: str) -> List[PerInstanceStats]:
+    per_instance_stats_path: str = os.path.join(run_path, "per_instance_stats.json")
+    if not os.path.exists(per_instance_stats_path):
+        raise ValueError(f"Could not load PerInstanceStats from {per_instance_stats_path}")
+    with open(per_instance_stats_path) as f:
+        raw_per_instance_stats = json.load(f)
+        return [dacite.from_dict(PerInstanceStats, r, config=_DACITE_CONFIG) for r in raw_per_instance_stats]
+
+
+def _truncate_predicted_text(
+    predicted_text: str, request_state: RequestState, adapter_spec: AdapterSpec
+) -> Optional[str]:
+    method = adapter_spec.method
+    prefix = ""
+    if method.startswith("multiple_choice_separate_"):
+        prefix = request_state.instance.input
+    elif method == "language_modeling":
+        if request_state.result is not None and request_state.result.completions:
+            tokens = request_state.result.completions[0].tokens
+            if tokens:
+                first_token = tokens[0]
+                if not first_token.top_logprobs:
+                    prefix = first_token.text
+    if prefix:
+        predicted_text = predicted_text.strip()
```

Comment:
Perhaps we don't need all these strips?

Reply:
From my (limited) manual testing, it seems to work without strips, so I removed them.